$MirOS: ports/devel/glib/patches/patch-glib_gmacros_h,v 1.2 2005/09/09 10:39:26 bsiegert Exp $
$OpenBSD: patch-glib_gmacros_h,v 1.3 2004/06/22 03:15:02 marcm Exp $
--- glib/gmacros.h.orig	Sun Dec  4 06:41:35 2005
+++ glib/gmacros.h	Thu Mar  6 20:22:39 2008
@@ -34,6 +34,7 @@
 /* We include stddef.h to get the system's definition of NULL
  */
 #include <stddef.h>
+#include <sys/param.h>
 
 /* Here we provide G_GNUC_EXTENSION as an alias for __extension__,
  * where this is valid. This allows for warningless compilation of
@@ -163,11 +164,13 @@
 #define	TRUE	(!FALSE)
 #endif
 
-#undef	MAX
+#ifndef MAX
 #define MAX(a, b)  (((a) > (b)) ? (a) : (b))
+#endif
 
-#undef	MIN
+#ifndef MIN
 #define MIN(a, b)  (((a) < (b)) ? (a) : (b))
+#endif
 
 #undef	ABS
 #define ABS(a)	   (((a) < 0) ? -(a) : (a))
